src/test/java/org/apache/commons/io/IOUtilsTestCase.java

```diff
@@ -311,28 +314,38 @@ public synchronized void close() throws IOException {
             assertTrue(IOUtils.contentEqualsIgnoreEOL(input1, input1));
         }
 
-        Reader r1;
-        Reader r2;
-
-        r1 = new CharArrayReader("".toCharArray());
-        r2 = new CharArrayReader("".toCharArray());
-        assertTrue(IOUtils.contentEqualsIgnoreEOL(r1, r2));
-
-        r1 = new CharArrayReader("1".toCharArray());
-        r2 = new CharArrayReader("1".toCharArray());
-        assertTrue(IOUtils.contentEqualsIgnoreEOL(r1, r2));
-
-        r1 = new CharArrayReader("1".toCharArray());
-        r2 = new CharArrayReader("2".toCharArray());
-        assertFalse(IOUtils.contentEqualsIgnoreEOL(r1, r2));
-
-        r1 = new CharArrayReader("123\rabc".toCharArray());
-        r2 = new CharArrayReader("123\nabc".toCharArray());
-        assertTrue(IOUtils.contentEqualsIgnoreEOL(r1, r2));
-
-        r1 = new CharArrayReader("321".toCharArray());
-        r2 = new CharArrayReader("321\r\n".toCharArray());
-        assertTrue(IOUtils.contentEqualsIgnoreEOL(r1, r2));
+        testSingleEOL("", "", true);
+        testSingleEOL("", "\n", true);
+        testSingleEOL("", "\r", true);
+        testSingleEOL("", "\r\n", true);
+        testSingleEOL("", "\r\r", false);
+        testSingleEOL("", "\n\n", false);
+        testSingleEOL("1", "1", true);
+        testSingleEOL("1", "2", false);
+        testSingleEOL("123\rabc", "123\nabc", true);
+        testSingleEOL("321", "321\r\n", true);
+        testSingleEOL("321", "321\r\naabb", false);
+        testSingleEOL("321", "321\n", true);
+        testSingleEOL("321", "321\r", true);
+        testSingleEOL("321", "321\r\n", true);
+        testSingleEOL("321", "321\r\r", false);
+        testSingleEOL("321", "321\n\r", false);
+        testSingleEOL("321\n", "321", true);
+        testSingleEOL("321\n", "321\n\r", false);
+        testSingleEOL("321\n", "321\r\n", true);
+        testSingleEOL("321\r", "321\r\n", true);
+        testSingleEOL("321\r\n", "321\r\n\r", false);
+        testSingleEOL("123", "1234", false);
+    }
+
```

Comment:
Please don't make unrelated changes to test code.

Reply:
> Please don't make unrelated changes to test code.

@sebbASF but the original tests are too loose, it did not cover some situations.
in other words, codes which can pass the original tests might be wrong.
So I added some test codes.
If you want me to move them to another file, I'll do it.

Reply:
Issues with existing tests are should be handled by a separate PR.
For new code, I would only expect to see new tests.

Reply:
@sebbASF OK, I will create a new pr for the changes IOUtilsTestCase.java

Reply:
@sebbASF 
pr created at
https://github.com/apache/commons-io/pull/137